### DISCUSSION TITLE
Changing branches not reflected when I do `git_branch`

### DISCUSSION THREAD

**Discussion by cemore2048** [fastlane/fastlane#19895] [Q&A]
In one of my lanes I create and switch to the new branch by using "git checkout -b branchB" 

but in my next lane when I do `git_branch` it looks like it's still thinking it's on branchA.

From my understanding of the [docs](http://docs.fastlane.tools/actions/git_branch/#lane-variables) there are some `GIT_BRANCH_ENV_VARS ` env variables that are setup that assume we're on the initial branch we ran the Fastline script on, but they don't update when we switch branches within our lanes.

Is there a way to switch branches and have `git_branch` reflect that?

**Comment by AlvaroBrey** [comment]
Years later this situation still happens when running in CI. It's because `git_branch()` inexplicably ignores the actual checked out branch in favor of CI variables ([source](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/helper/git_helper.rb#L127))

To solve this you need to set the `FL_GIT_BRANCH_DONT_USE_ENV_VARS` env variable.
